frontend/src/FirstRun/AuthenticationRequiredModalContent.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Alert from 'Components/Alert';
+import FormGroup from 'Components/Form/FormGroup';
+import FormInputGroup from 'Components/Form/FormInputGroup';
+import FormLabel from 'Components/Form/FormLabel';
+import SpinnerButton from 'Components/Link/SpinnerButton';
+import LoadingIndicator from 'Components/Loading/LoadingIndicator';
+import ModalBody from 'Components/Modal/ModalBody';
+import ModalContent from 'Components/Modal/ModalContent';
+import ModalFooter from 'Components/Modal/ModalFooter';
+import ModalHeader from 'Components/Modal/ModalHeader';
+import { inputTypes, kinds } from 'Helpers/Props';
+import {
+  authenticationMethodOptions,
+  authenticationRequiredOptions,
+} from 'Settings/General/SecuritySettings';
+import { clearPendingChanges } from 'Store/Actions/baseActions';
+import {
+  fetchGeneralSettings,
+  saveGeneralSettings,
+  setGeneralSettingsValue,
+} from 'Store/Actions/settingsActions';
+import { fetchStatus } from 'Store/Actions/systemActions';
+import createSettingsSectionSelector from 'Store/Selectors/createSettingsSectionSelector';
+import translate from 'Utilities/String/translate';
+import styles from './AuthenticationRequiredModalContent.css';
+
+const SECTION = 'general';
+
+const selector = createSettingsSectionSelector(SECTION);
+
+function onModalClose() {
+  // No-op
+}
+
+export default function AuthenticationRequiredModalContent() {
+  const { isPopulated, error, isSaving, settings } = useSelector(selector);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchGeneralSettings());
+  }, [dispatch]);
+
+  useEffect(() => {
+    return () => {
+      dispatch(clearPendingChanges());
+    };
+  }, [dispatch]);
```

Comment:
Should be safe to do this as a single `useEffect` hook

```suggestion
  useEffect(() => {
    dispatch(fetchGeneralSettings());
    
    return () => {
      dispatch(clearPendingChanges());
    };
  }, [dispatch]);
```